### PR TITLE
fix(trace): Handle multiple perf issues in a trace

### PR DESCRIPTION
**Pull request by wmak**
- This fixes a bug where we weren't handling multiple perf issues in a single trace since we did it by mapping event_ids to occurrence_ids, but there could be multiple occurrence_id for a given event_id
- This also fixes a bug (only with new spans view) where the list of suspect spans was _all_ the suspect spans across all perf issues for every perf issue, rather than just the suspect spans for a given perf issue
- Snuck some decorators to add more tracing too

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 1]
```diff
@@ -189,6 +192,64 @@ def nodestore_event(self) -> Event | None:
...
+                            end = end_timestamp
+                        else:
+                            end = max(end, end_timestamp)
+                        suspect_spans.append(event_span.get("span_id"))
+            # Logic for qualified_short_id is copied from property on the Group model
+            # to prevent an N+1 query from accessing project.slug everytime
+            qualified_short_id = None
+            project_slug = self.event["project"]
+            if group.short_id is not None:
+                qualified_short_id = f"{project_slug.upper()}-{base32_encode(group.short_id)}"
```
Comment (k-fish) [inline src/sentry/api/endpoints/organization_events_trace.py:233]: So we know that this project slug is unique per this `TraceEvent`, maybe instead allow `qualified_short_id` on the Group model to be passed a project_id to skip n+1, and mention this on the model instead so people can benefit in the future?

**Review by k-fish — APPROVED** [review]
You have a few tests that a breaking, but seems related to `suspect_spans` as mentioned, once those are fixed, lgtm

File: src/sentry/api/endpoints/organization_events_trace.py (hunk 1 above)
Comment (wmak) [reply src/sentry/api/endpoints/organization_events_trace.py]: I'll do this in a follow up

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 2]
```diff
@@ -189,6 +192,64 @@ def nodestore_event(self) -> Event | None:
...
+                    "project_id": self.event["project.id"],
+                    "project_slug": self.event["project"],
+                    "title": group.title,
+                    "level": constants.LOG_LEVELS[group.level],
+                    "culprit": group.culprit,
+                    "type": group.type,
+                    "start": start,
+                    "end": end,
+                }
+            )
```
[the remark this answers is not in the document]
Comment (wmak) [reply src/sentry/api/endpoints/organization_events_trace.py]: as discussed in slack, looks like this is completely wrong lol. Re did my entire approach :+1:

**Pull request merged**